src/ai/ai.cpp

```diff
@@ -857,20 +859,36 @@ void get_desired_factory_types(sys::state& state, dcon::nation_id nid, std::vect
 			}
 		} // END if building unlocked
 	}
+	*/
+
 	if(desired_types.empty()) { // second pass: try to make money
 		for(auto type : state.world.in_factory_type) {
 			if(n.get_active_building(type) || type.get_is_available_from_start()) {
 				auto& inputs = type.get_inputs();
 				bool lacking_input = false;
+				bool lacking_output = n.get_demand_satisfaction(type.get_output()) < 1.02f;
```

Comment:
unless you have changed something, demand satisfaction should always be in the range 0 - 1 (representing percent of the demand that could be purchased)